### PR TITLE
8286191: misc tests fail due to JDK-8285987

### PR DESCRIPTION
**Pull request by MBaesken**
The isMusl method had to be handled in test/lib-test/jdk/test/lib/TestMutuallyExclusivePlatformPredicates.java .
Additionally, the vm.musl predicate seem not to be available in the langtools tests.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286191](https://bugs.openjdk.java.net/browse/JDK-8286191): misc tests fail due to JDK-8285987

### Reviewers
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**) ⚠️ Review applies to [2337301c](https://git.openjdk.java.net/jdk/pull/8556/files/2337301cbc9eef8a7ebf081f2b897a0a8120aa29)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8556/head:pull/8556` \
`$ git checkout pull/8556`

Update a local copy of the PR: \
`$ git checkout pull/8556` \
`$ git pull https://git.openjdk.java.net/jdk pull/8556/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8556`

View PR using the GUI difftool: \
`$ git pr show -t 8556`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8556.diff">https://git.openjdk.java.net/jdk/pull/8556.diff</a>

</details>

### CODE REVIEW

**Review by RogerRiggs — APPROVED** [review]
LGTM

**Review by dcubed-ojdk — CHANGES_REQUESTED** [review]

**Comment by dcubed-ojdk** [comment]
@MBaesken - I corrected a typo in my synopsis update for JDK-8286191.
Please use "/issue JDK-8286191" to update the PR's synopsis.

File: test/langtools/jdk/jshell/ExternalEditorTest.java [hunk 1]
```diff
@@ -25,8 +25,6 @@
  * @test
  * @summary Testing external editor.
  * @bug 8143955 8080843 8163816 8143006 8169828 8171130 8162989 8210808
- * @comment musl/Alpine has problems executing some shell scripts, see 8285987
- * @requires !vm.musl
```
[the remark this answers is not in the document]
Comment (MBaesken) [reply test/langtools/jdk/jshell/ExternalEditorTest.java:29]: Hi Daniel, I restored the original year 2017.
(additionally I removed the langtools exclusion)

> Presumably this "@requires" was added for some reason so what's going to happen if this test is run on Alpine Linux

I can only speak about our Alpine setup (3.15 in a container) and there the test fails with   error=8, Exec format error. Looks like something similar has been observed as well by these people
https://www.openwall.com/lists/musl/2018/03/09/2
https://github.com/scala-steward-org/scala-steward/issues/1374

**Comment by MBaesken** [comment]
> ...  update the PR's synopsis.
Done .

**Review by RogerRiggs — APPROVED** [review]
Looks good.
Thanks for resolving both ProblemLists.

Hopefully, the real problem and solution on Musl can be found separately.

**Pull request closed without merging**